backend/infrahub/message_bus/operations/send/telemetry.py

```diff
@@ -111,21 +111,23 @@ async def push(
         "data": data,
         "checksum": hashlib.sha256(json.dumps(data).encode()).hexdigest(),
     }
-
-    async with httpx.AsyncClient(timeout=5.0) as client:
-        try:
-            response = await client.post(config.SETTINGS.main.telemetry_endpoint, json=payload)
-            response.raise_for_status()
-        except httpx.HTTPError as exc:
-            service.log.debug(f"HTTP exception while pushing anonymous telemetry: {exc}")
+    try:
+        response = await service.http.post(url=config.SETTINGS.main.telemetry_endpoint, json=payload)
+    except HTTPServerError as exc:
+        service.log.debug(f"HTTP exception while pushing anonymous telemetry: {exc}")
+    if not response.is_success:
+        service.log.debug("HTTP exception while pushing anonymous telemetry", status_code=response.status_code)
 
 
 @task(retries=5)
-async def post_telemetry_data(service: InfrahubServices, url: str, payload: dict[str, Any]) -> None:  # pylint: disable=unused-argument
+async def post_telemetry_data(service: InfrahubServices, url: str, payload: dict[str, Any]) -> None:
     """Send the telemetry data to the specified URL, using HTTP POST."""
-    async with httpx.AsyncClient() as client:
-        response = await client.post(url=url, json=payload)
-        response.raise_for_status()
+    try:
+        response = await service.http.post(url=url, json=payload)
+    except HTTPServerError as exc:
+        service.log.debug(f"HTTP exception while pushing anonymous telemetry: {exc}")
+    if not response.is_success:
```

Comment:
ideally we should raise an exception here as well and let prefect handle the retry

Reply:
I've updated this so that an exception would be raised.